core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java

```diff
@@ -237,4 +248,23 @@ public static String jobName(RunContext runContext) {
         String suffix = RandomStringUtils.randomAlphanumeric(5).toLowerCase();
         return normalized + "-" + suffix;
     }
+
+    private static String getSeparator(TargetOS targetOS) {
+        switch (targetOS) {
+            case TargetOS.LINUX -> {
+                return "\n";
+            }
+            case TargetOS.WINDOWS -> {
+                return "\r\n";
+            }
+            default -> {
+                return System.lineSeparator();
+            }
+        }
+    }
```

Comment:
You can add the line separator as an enumeration parameter so you can remove this method and call targetOs.lineSeparator() instead.